src/unittests/maths.spec.ts

```diff
@@ -38,8 +42,30 @@ g.test('test,math,diffULP')
     { a: -hexToF32(0x00800001), b: -hexToF32(0x00800000), ulp: 1 }, // Single mantissa step
     { a: -hexToF32(0x03800000), b: -hexToF32(0x03800001), ulp: 1 }, // Single mantissa step
     { a: -hexToF32(0x03800001), b: -hexToF32(0x03800000), ulp: 1 }, // Single mantissa step
-    { a: hexToF32(0x00800000), b: -hexToF32(0x00800000), ulp: 2 ** 24 }, // Two exponent steps
-    { a: -hexToF32(0x00800000), b: hexToF32(0x00800000), ulp: 2 ** 24 }, // Two exponent steps
+    { a: hexToF32(0x00800000), b: hexToF32(0x00800002), ulp: 2 }, // Double mantissa step
+    { a: hexToF32(0x00800002), b: hexToF32(0x00800000), ulp: 2 }, // Double mantissa step
+    { a: hexToF32(0x03800000), b: hexToF32(0x03800002), ulp: 2 }, // Double mantissa step
+    { a: hexToF32(0x03800002), b: hexToF32(0x03800000), ulp: 2 }, // Double mantissa step
+    { a: -hexToF32(0x00800000), b: -hexToF32(0x00800002), ulp: 2 }, // Double mantissa step
+    { a: -hexToF32(0x00800002), b: -hexToF32(0x00800000), ulp: 2 }, // Double mantissa step
+    { a: -hexToF32(0x03800000), b: -hexToF32(0x03800002), ulp: 2 }, // Double mantissa step
+    { a: -hexToF32(0x03800002), b: -hexToF32(0x03800000), ulp: 2 }, // Double mantissa step
+    { a: hexToF32(0x00800000), b: 0, ulp: 1 }, // Normals near 0
+    { a: 0, b: hexToF32(0x00800000), ulp: 1 }, // Normals near 0
+    { a: -hexToF32(0x00800000), b: 0, ulp: 1 }, // Normals near 0
+    { a: 0, b: -hexToF32(0x00800000), ulp: 1 }, // Normals near 0
+    { a: hexToF32(0x00800000), b: -hexToF32(0x00800000), ulp: 2 }, // Normals around 0
+    { a: -hexToF32(0x00800000), b: hexToF32(0x00800000), ulp: 2 }, // Normals around 0
+    { a: hexToF32(0x00000001), b: 0, ulp: 0 }, // Subnormals near 0
+    { a: 0, b: hexToF32(0x00000001), ulp: 0 }, // Subnormals near 0
+    { a: -hexToF32(0x00000001), b: 0, ulp: 0 }, // Subnormals near 0
+    { a: 0, b: -hexToF32(0x00000001), ulp: 0 }, // Subnormals near 0
+    { a: hexToF32(0x00000001), b: -hexToF32(0x00000001), ulp: 0 }, // Subnormals near 0
+    { a: -hexToF32(0x00000001), b: hexToF32(0x00000001), ulp: 0 }, // Subnormals near 0
+    { a: hexToF32(0x00000001), b: hexToF32(0x00800000), ulp: 1 }, // Normal/Subnormal boundary
+    { a: hexToF32(0x00800000), b: hexToF32(0x00000001), ulp: 1 }, // Normal/Subnormal boundary
+    { a: -hexToF32(0x00000001), b: -hexToF32(0x00800000), ulp: 1 }, // Normal/Subnormal boundary
+    { a: -hexToF32(0x00800000), b: -hexToF32(0x00000001), ulp: 1 }, // Normal/Subnormal boundary
```

Comment:
Please add cases that are slightly away from the edge cases.

I used these to convince myself (finally!) that the subtraction was the right thing to do:

    { a: hexToF32(0x00800001), b: hexToF32(0x00000000), ulp: 2 }, // Just-above-Normal/Subnormal boundary
    { a: hexToF32(0x00800001), b: hexToF32(0x00000001), ulp: 2 }, // Just-above-Normal/Subnormal boundary
    { a: hexToF32(0x00800005), b: hexToF32(0x00000001), ulp: 6 }, // Just-above-Normal/Subnormal boundary
    { a: hexToF32(0x00800005), b: hexToF32(0x00000111), ulp: 6 }, // Just-above-Normal/Subnormal boundary

Reply:
Done